core/templates/dev/head/components/create_exploration_button_directive.html

```diff
@@ -0,0 +1,27 @@
+<script type="text/ng-template" id="components/createExplorationButton">
+
+  <div class="pull-right oppia-navbar-button-container">
+    {% if not has_fully_registered %}
+      <!-- User must complete the registration process. -->
+      <a class="btn oppia-navbar-add-exploration-button protractor-test-create-exploration oppia-transition-200"
+         href ng-click="onRedirectToLogin('{{gallery_login_redirect_url}}')">
```

Comment:
Ooh, this looks a bit problematic. Whether {{...}} gets jinja-interpolated depends on which page this is on, and there's a good chance it'll fail silently if this directive gets used in more places in the future.

For now, I suggest just replacing this with a direct link to `/my_explorations?mode=create`. This means that, if a non-logged-in user clicks this button, they will be taken to that page -- and then be required to log in, since a login is required to reach that page; after they log in, they'll see the creation modal and can continue creating. This has the advantage that it will work automatically if the "create exploration" button is put on all pages. If you agree with this, I think it would also be good to get rid of gallery_login_redirect_url altogether, too (and replace other instances similarly, if necessary).